File: VB/TextEditPasswordStrengthBar/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms

Namespace TextEditPasswordStrengthBar

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub textEditPwdStrength_Leave(ByVal sender As Object, ByVal e As EventArgs)
            Dim te As TextEditPwdStrength = TryCast(sender, TextEditPwdStrength)
            te.Properties.PasswordChar = "*"c
        End Sub

        Private Sub textEditPwdStrength_Enter(ByVal sender As Object, ByVal e As EventArgs)
            Dim te As TextEditPwdStrength = TryCast(sender, TextEditPwdStrength)
            te.Properties.PasswordChar = Microsoft.VisualBasic.Strings.ChrW(0)
        End Sub
    End Class
End Namespace

File: CS/TextEditPasswordStrengthBar/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace TextEditPasswordStrengthBar
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        void textEditPwdStrength_Leave(object sender, EventArgs e)
        {
            TextEditPwdStrength te = sender as TextEditPwdStrength;
            te.Properties.PasswordChar = '*';
        }

        void textEditPwdStrength_Enter(object sender, EventArgs e)
        {
            TextEditPwdStrength te = sender as TextEditPwdStrength;
            te.Properties.PasswordChar = '\0';
        }
    }
}
